### DISCUSSION TITLE
Which is more accurate? The reported power output, or the reported voltage/current?

### DISCUSSION THREAD

**Discussion by lenwar** [StephanJoubert/home_assistant_solarman#303] [Q&A] [answered]
Hi,

I have a number of Deye-something-something microinverters. (Using the deye-2mppt.yaml )
That yaml-file provides a lot of output.

The Voltage and Current of both connected solar panels

sensor.solarman1_pv1_voltage
sensor.solarman1_pv1_current
And
sensor.solarman1_pv2_voltage
sensor.solarman1_pv2_current

But also:
sensor.solarman1_total_ac_output_power_active

There however is a significant difference (0.5%-2%) between those values:

```
{{ states('sensor.solarman1_total_ac_output_power_active') }} ## (( at moment of writing 472 ))

{{
 (float(states('sensor.solarman1_pv1_current')) * float(states('sensor.solarman1_pv1_voltage'))) + 
 (float(states('sensor.solarman1_pv2_current')) * float(states('sensor.solarman1_pv2_voltage'))) ## (( at moment of writing about 468.57 ))
}}
```
At the moment of writing the difference is about 3,4 Watt, which is almost 1% (I have seen larger and smaller differences between the given values, anything from almost the same to little more than 2%)

Does anyone know which value is more accurate in practice? The provided output power of the inverter, or the reported voltage/current. Has anyone ever looked in to this??

**Comment by helgard999** [comment]
I would speculate that multiplying Current and Voltage gives Apparent
Power. The % difference is the Apparent Power compared to the Active Power.
With a bit of trigonometry, you would be able to calculate the phase angle
load, I would assume that I and V from the PV are in phase, so the
difference must be the phase angle introduced by the load.

On Sun, May 21, 2023 at 3:34 PM lenwar ***@***.***> wrote:

> Hi,
>
> I have a number of Deye-something-something microinverters. (Using the
> deye-2mppt.yaml )
> That yaml-file provides a lot of output.
>
> The Voltage and Current of both connected solar panels
>
> sensor.solarman1_pv1_voltage
> sensor.solarman1_pv1_current
> And
> sensor.solarman1_pv2_voltage
> sensor.solarman1_pv2_current
>
> But also:
> sensor.solarman1_total_ac_output_power_active
>
> There however is a significant difference (0.5%-2%) between those values:
>
> {{ states('sensor.solarman1_total_ac_output_power_active') }} ## (( at moment of writing 472 ))
>
> {{
>  (float(states('sensor.solarman1_pv1_current')) * float(states('sensor.solarman1_pv1_voltage'))) +
>  (float(states('sensor.solarman1_pv2_current')) * float(states('sensor.solarman1_pv2_voltage'))) ## (( at moment of writing about 468.57 ))
> }}
>
> At the moment of writing the difference is about 3,4 Watt, which is almost
> 1% (I have seen larger and smaller differences between the given values,
> anything from almost the same to little more than 2%)
>
> Does anyone know which value is more accurate in practice? The provided
> output power of the inverter, or the reported voltage/current. Has anyone
> ever looked in to this??
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/StephanJoubert/home_assistant_solarman/discussions/303>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMR3RL76URKQP3LURSJ5243XHIKT3ANCNFSM6AAAAAAYJNBTIQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Groete
Helgard Ross
+27 83 459 4052

**Comment by hagagit** [comment, marked as answer]
The module data (pv1/pv2) is the input power coming from the panels. The ac value is the output power of the inverter. The difference is the eficency of the inverter. Would be interesting to plot the difference over the output power. That would give an efficency curve of the inverter. 
So both values seem to be accurate, just showing different things.

**Reply by lenwar** [reply]
That makes sense.
Thanks. I'll make a chart based on the difference. Should be interesting. Also how steady the efficiency is. (At 10%, 20%, .... load)

**Reply by lenwar** [reply]
Quick question though.
The values I get don't make sense to me (my physics is at High School level by the way)

The input values are higher than the output levels.

At moment of writing (at the end of the day) the combined inputs are at approx 170VA and the outputs are 180W combined

(( over the entire day, the output is higher than the input ))

**Reply by hagagit** [reply]
I see this in my inverter aswell. It seems these measurements are not that accurate.
